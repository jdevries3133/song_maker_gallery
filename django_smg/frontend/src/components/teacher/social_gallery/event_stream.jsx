import React, { useRef } from "react";
import PropTypes from "prop-types";

import styled from "Styles";
import { useScrollCallback } from "Common/useScrollCallback";

import { SocialEvent } from "./event";

const Container = styled.div`
  max-width: 100vw;
  max-height: 100vh;
  overflow: scroll;
`;

export const EventStream = ({ events, fetchMoreEvents }) => {
  const rootRef = useRef(null);
  const targetRef = useRef(null);

  useScrollCallback(fetchMoreEvents, targetRef, rootRef);

  return (
    <Container ref={rootRef}>
      {events.map((event, i) =>
        // target second to last item for infinite scroll trigger
        i == events.length - 2 ? (
          <SocialEvent key={i} ref={targetRef} event={event} />
        ) : (
          <SocialEvent key={i} event={event} />
        )
      )}
    </Container>
  );
};

EventStream.propTypes = {
  events: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.oneOf(["like", "heart", "star", "comment"]),
      username: PropTypes.string,

      /* contains comment text if applicable */
      value: PropTypes.string,
      /* for comment pre-approval */
      isApproved: PropTypes.bool,

      /* same entity as the song passed to <DynamicTile /> */
      song: PropTypes.object,
    })
  ),

  /* callback for when user has scrolled close to the bottom */
  fetchMoreEvents: PropTypes.func,
};
