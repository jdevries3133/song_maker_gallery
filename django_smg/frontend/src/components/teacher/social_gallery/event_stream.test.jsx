import React from "react";
import renderer from "react-test-renderer";

import { EVENTS } from "./__mocks__/data.js";
import { EventStream } from "./event_stream";

it("matches snapshot", () => {
  expect(
    renderer.create(<EventStream events={EVENTS} />).toJSON()
  ).toMatchSnapshot();
});
