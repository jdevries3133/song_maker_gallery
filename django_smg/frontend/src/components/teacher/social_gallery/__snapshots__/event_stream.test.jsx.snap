// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-hBURRC bIvzMU"
>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        üëç
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Janice
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚ù§Ô∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          James
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚≠êÔ∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Asaun
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚úçÔ∏è
      </span>
    </div>
    <div
      className="sc-egiSv jfUQsi"
    >
      <h4
        className="styles__H4-ggNJUq sc-bqiQRQ  kgtDui"
      >
        <b>
          Mike
        </b>
         commented on 
        <b>
          Josiah
          's
        </b>
         song
      </h4>
      <p
        className="sc-ksdxAp ZViNI"
      >
        Wohoo! üòÅ üòÅ
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
    <form
      className="sc-dkPtyc fuaMj"
      onSubmit={[Function]}
    >
      <label
        className="sc-hKwCoD dhnZkD"
        htmlFor="approve"
      >
        Approve Comment?
      </label>
      <input
        className="styles__Button-dxTSuI sc-eCImvq kieHs daNQhM"
        id="approve"
        type="submit"
        value="Approve"
      />
    </form>
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚ù§Ô∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Janice
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚úçÔ∏è
      </span>
    </div>
    <div
      className="sc-egiSv jfUQsi"
    >
      <h4
        className="styles__H4-ggNJUq sc-bqiQRQ  kgtDui"
      >
        <b>
          Amir
        </b>
         commented on 
        <b>
          Josiah
          's
        </b>
         song
      </h4>
      <p
        className="sc-ksdxAp ZViNI"
      >
        Good stuff my guy
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚ù§Ô∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Janice
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚úçÔ∏è
      </span>
    </div>
    <div
      className="sc-egiSv jfUQsi"
    >
      <h4
        className="styles__H4-ggNJUq sc-bqiQRQ  kgtDui"
      >
        <b>
          Amir
        </b>
         commented on 
        <b>
          Josiah
          's
        </b>
         song
      </h4>
      <p
        className="sc-ksdxAp ZViNI"
      >
        Good stuff my guy
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚ù§Ô∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Janice
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚úçÔ∏è
      </span>
    </div>
    <div
      className="sc-egiSv jfUQsi"
    >
      <h4
        className="styles__H4-ggNJUq sc-bqiQRQ  kgtDui"
      >
        <b>
          Shawn
        </b>
         commented on 
        <b>
          Josiah
          's
        </b>
         song
      </h4>
      <p
        className="sc-ksdxAp ZViNI"
      >
        That is all fine and good, but did you think about what would happen if you ended up with a really, really long comment in here?
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
    <form
      className="sc-dkPtyc fuaMj"
      onSubmit={[Function]}
    >
      <label
        className="sc-hKwCoD dhnZkD"
        htmlFor="approve"
      >
        Approve Comment?
      </label>
      <input
        className="styles__Button-dxTSuI sc-eCImvq kieHs daNQhM"
        id="approve"
        type="submit"
        value="Approve"
      />
    </form>
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚ù§Ô∏è
      </span>
    </div>
    <div
      className="sc-iqsfdx kQKKGL"
    >
      <p
        className="sc-crHlIS fiQlej"
      >
        <b>
          Janice
        </b>
         reacted to 
        <b>
          Josiah
          's
        </b>
         song
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
  </div>
  <div
    className="sc-jrQzUz cTczIb"
  >
    <div
      className="sc-kDThTU bdnIYs"
    >
      <span
        className="icon__Span-jtAmdl IYVOc"
      >
        ‚úçÔ∏è
      </span>
    </div>
    <div
      className="sc-egiSv jfUQsi"
    >
      <h4
        className="styles__H4-ggNJUq sc-bqiQRQ  kgtDui"
      >
        <b>
          Amir
        </b>
         commented on 
        <b>
          Josiah
          's
        </b>
         song
      </h4>
      <p
        className="sc-ksdxAp ZViNI"
      >
        Good stuff my guy
      </p>
    </div>
    <svg
      className="song_preview__StyledPreview-jHaHVD jcxRRe"
      viewBox="0 0 100 100"
    />
    <form
      className="sc-dkPtyc fuaMj"
      onSubmit={[Function]}
    >
      <label
        className="sc-hKwCoD dhnZkD"
        htmlFor="approve"
      >
        Approve Comment?
      </label>
      <input
        className="styles__Button-dxTSuI sc-eCImvq kieHs daNQhM"
        id="approve"
        type="submit"
        value="Approve"
      />
    </form>
  </div>
</div>
`;
